src/main/java/Models/Appointment/Appointment.java

```diff
@@ -0,0 +1,286 @@
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package Models.Appointment;
+
+import java.sql.SQLException;
+import java.sql.Time;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.servlet.RequestDispatcher;
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ *
+ * @author maxwell
+ */
+public class Appointment {
+
+    public ArrayList<Event> sortAppointmentsByDate(String appointmentDate, ArrayList<Event> event) {
+        ArrayList<Event> e = new ArrayList<>();
+
+        event.stream().peek((Event event1) -> System.out.println(event1.getDate())).filter((event1) -> (event1.getDate().equals(appointmentDate))).forEachOrdered(e::add);
+        return e;
+    }
+
+    public ArrayList<String> getGeneratedSlots(int apptmLength, String doctorName, String date) {
+        ArrayList<Event> currEvents;
+        ArrayList<String> intervals = new ArrayList<>();
+        try {
+            String[] hours = new String[2];
+            boolean addSlot;
+            SimpleDateFormat timeFormat = new SimpleDateFormat("HH:mm");
+
+            currEvents = AppointmentDAO.retrieveDoctorsAppointment(doctorName, true);
+            currEvents = sortAppointmentsByDate(date, currEvents);
+
+            Calendar cal = Calendar.getInstance();
+            Calendar cal2 = Calendar.getInstance();
+            cal.setTime(timeFormat.parse("9:00"));
+            cal2.setTime(timeFormat.parse("18:00"));
+            Time u = new java.sql.Time(cal.getTimeInMillis());
+            hours[0] = timeFormat.format(u);
+
+            while (cal.getTime().before(cal2.getTime())) {
+                cal.add(Calendar.MINUTE, apptmLength);
+                Time startTime = new java.sql.Time(cal.getTimeInMillis());
+                String x = timeFormat.format(startTime);
+                hours[1] = x;
+                addSlot = isAvailableSlot(hours, currEvents);
+                if (addSlot) {
+                    String hoursRange = hours[0] + '-' + hours[1];
+                    intervals.add(hoursRange);
+                }
+                String temp = hours[1];
+                hours[0] = temp;
+            }
+        } catch (ParseException | SQLException e) {
+            Logger.getLogger(Appointment.class.getName()).log(Level.SEVERE, null, e);
+        }
+        return intervals;
+    }
+
+    public boolean isAvailableSlot(String[] time, ArrayList<Event> currEvents) {
+
+        int slotStart = Integer.parseInt(time[0].replace(":", ""));
+        int slotEnd = Integer.parseInt(time[1].replace(":", ""));
+        for (Event currEvent : currEvents) {
+            int appointmentstart = Integer.parseInt(currEvent.getStartTime().replace(":", ""));
+            int appointmentend = Integer.parseInt(currEvent.getEndTime().replace(":", ""));
```

Comment:
There are a bunch of formatting / naming comments I could make in this PR, but I think in the interest of time I'll refrain for now and pick it up in a quick PR at the end of this sprint or next sprint

Reply:
Yeah, we'll run a cleanup PR after implementations are done I imagine - I fixed most of the linting issues in Max's code but didn't touch most variable names etc